jupyterhub/templates/proxy/deployment.yaml

```diff
@@ -15,22 +16,14 @@ spec:
         # This lets us autorestart when the secret changes!
         checksum/hub-secret: {{ include (print $.Template.BasePath "/hub/secret.yaml") . | sha256sum }}
         checksum/proxy-secret: {{ include (print $.Template.BasePath "/proxy/secret.yaml") . | sha256sum }}
-        hub.jupyter.org/https-type: {{ .Values.proxy.https.type }}
       labels:
         name: proxy
         component: proxy
         release: {{ .Release.Name }}
         heritage: {{ .Release.Service }}
-        {{- if eq .Values.proxy.https.type "letsencrypt" }}
-        # required for kube-lego to work
-        app: kube-lego
```

Comment:
We could add `app: jupyterhub` as a label now that this hack is out of the way :)

Reply:
I get a warm feeling about this PR! I don't grasp whats done fully, but it seems like you are successfully separating concerns and making the code more understandable!